src/angularComponent/ngcOmniboxController.js

```diff
@@ -106,17 +109,51 @@ export default class NgcOmniboxController {
    * @returns {Number} -- Index of the newly highlighted item
    */
   highlightNext() {
-    if (this.highlightedIndex < this._suggestionItems.length - 1) {
-      this.highlightedIndex++;
+    let newIndex = this.highlightedIndex;
+
+    if (newIndex < this._suggestionsUiList.length - 1) {
+      newIndex++;
     } else {
-      this.highlightedIndex = 0;
+      newIndex = 0;
     }
 
-    return this.highlightedIndex;
+    // Protect against an infinite loop in cases where all items are disabled
+    if (this.startHighlightIndex === newIndex) {
+      this.highlightNone();
+      return this.highlightedIndex;
+    } else {
+      this.highlightedIndex = newIndex;
+    }
+
+    const suggestion = this._suggestionsUiList[newIndex];
+    if (!this.isSelectable({suggestion: suggestion.data})) {
+      if (this.startHighlightIndex === null || typeof this.startHighlightIndex === 'undefined') {
+        this.startHighlightIndex = newIndex;
+      }
+
+      this.highlightNext();
+    } else {
+      this.startHighlightIndex = null;
+    }
+
+    return newIndex;
+  }
+
+  /**
+   * Un-highlights all suggestions.
+   */
+  highlightNone() {
+    this.highlightedIndex = -1; // -1 means nothing is highlighted
   }
 
   isHighlighted(item) {
-    return this._suggestionItems.indexOf(item) === this.highlightedIndex;
+    const match = this._suggestionsUiList.find((listItem) => listItem.data === item);
```

Comment:
How often is this method called? Perhaps it would be more performant to keep a reference to the currently highlighted data item attached to `this` all the time?